keyboards/singa/singa.c

```diff
@@ -34,6 +34,25 @@ void rgblight_set(void) {
 }
 #endif
 
+void matrix_init_kb(void) {
+#ifdef RGBLIGHT_ENABLE
+    if (rgblight_config.enable) {
+        i2c_init();
+        i2c_transmit(0xb0, (uint8_t*)led, 3 * RGBLED_NUM, 100);
+    }
+#endif
+    // call user level keymaps, if any
+    matrix_init_user();
+}
+
+void matrix_scan_kb(void) {
+#ifdef RGBLIGHT_ENABLE
+    rgblight_task();
```

Comment:
Is this actually needed?  It should be called automatically, IIRC.

Reply:
I'll retest again later tonight, but it's in a number of the other ps2avrgb boards' code (including the template) and it's the only explanation I currently have for the animations not working.

Reply:
Confirmed that removing this breaks animations. Is it because this is implemented as a custom RGB driver?

Reply:
rgblight_task() is called from tmk_core\protocol\lufa\lufa.c, so unless you are using that main loop, you don't get that call and have to call it yourself.

Reply:
Shoot, yeah.... that's something that should probably be fixed, in the near future then. :(